packages/FamilyTree/Reflection/FamilyRelationsAnalyzer.php

```diff
@@ -36,4 +64,100 @@ public function getChildrenOfClassReflection(ClassReflection $desiredClassReflec
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        if (! $scope instanceof Scope) {
+            return new PropertyType($varType, $propertyTypeNode);
+        }
+
+        /** @var ClassReflection $classReflection */
+        $classReflection = $scope->getClassReflection();
+        $ancestors = $classReflection->getAncestors();
+        $propertyName = $this->nodeNameResolver->getName($property);
+
+        $kindPropertyFetch = $property->isStatic()
+            ? StaticPropertyFetch::class
+            : PropertyFetch::class;
+
+        $className = $property->getAttribute(AttributeKey::CLASS_NAME);
+        foreach ($ancestors as $ancestor) {
+            $ancestorName = $ancestor->getName();
+            if ($ancestorName === $className) {
+                continue;
+            }
+
+            $fileName = $ancestor->getFileName();
+            if ($fileName === false) {
+                continue;
+            }
+            $fileContent = $this->smartFileSystem->readFile($fileName);
+            $nodes = $this->parser->parse($fileContent);
+
+            $objectType = new ObjectType($ancestor->getName());
+            $parentType = new ObjectType('PHPUnit\Framework\TestCase');
+            $isPHPUnitTest = $parentType->isSuperTypeOf($objectType)
+                ->yes();
+
+            if ($isPHPUnitTest) {
+                continue;
+            }
+
+            if ($this->isFilled((array) $nodes, $propertyName, $kindPropertyFetch)) {
+                $varType = new UnionType([$varType, new NullType()]);
+                $propertyTypeNode = $this->staticTypeMapper->mapPHPStanTypeToPhpParserNode(
+                    $varType,
+                    TypeKind::KIND_PROPERTY
+                );
+
+                return new PropertyType($varType, $propertyTypeNode);
+            }
+        }
+
+        return new PropertyType($varType, $propertyTypeNode);
+    }
+
+    /**
+     * @param Stmt[] $nodes
+     */
+    private function isFilled(array $nodes, string $propertyName, string $kindPropertyFetch): bool
+    {
+        return (bool) $this->betterNodeFinder->findFirst($nodes, function (Node $n) use (
+            $propertyName,
+            $kindPropertyFetch
+        ): bool {
+            if (! $n instanceof ClassMethod) {
+                return false;
+            }
+
+            if ($this->nodeNameResolver->isNames($n->name, ['autowire', 'setUp'])) {
+                return false;
+            }
+
+            return (bool) $this->betterNodeFinder->findFirst((array) $n->stmts, function (Node $n2) use (
+                $propertyName,
+                $kindPropertyFetch
+            ): bool {
+                if (! $n2 instanceof Assign) {
+                    return false;
+                }
+
+                return $kindPropertyFetch === $n2->var::class && $this->nodeNameResolver->isName(
+                    $n2->var,
+                    $propertyName
+                );
```

Comment:
Could you extract this to own method? This seems rather complex to read

Reply:
implemented https://github.com/rectorphp/rector-src/pull/295/commits/bca88fc540314faa0e7ff84cfd4aa86f439364f6